File: core/src/kr/edoli/imview/ui/ColorActor.java
package kr.edoli.imview.ui;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.scenes.scene2d.Actor;
import kr.edoli.imview.ui.res.Textures;

/**
 * Created by daniel on 16. 2. 27.
 */
public class ColorActor extends Actor {
    private Color color;

    public ColorActor(Color color) {
        this.color = color;
    }

    @Override
    public void draw(Batch batch, float parentAlpha) {

        Color preColor = batch.getColor();

        batch.setColor(color);
        batch.draw(Textures.White, getX(), getY(), getWidth(), getHeight());

        batch.setColor(preColor);
    }
}

File: core/src/kr/edoli/imview/ui/ColorActor.kt
package kr.edoli.imview.ui

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.scenes.scene2d.Actor

/**
 * Created by daniel on 16. 9. 23.
 */
class ColorActor : Actor {

    constructor(color: Color) : super() {
        this.color = color
    }

    override fun draw(batch: Batch, parentAlpha: Float) {
        val c = batch.color

        batch.color = color
        batch.drawRect(x, y, width, height)

        batch.color = c
    }
}